jogo-oito/src/main/java/model/Matrix.java

```diff
@@ -12,69 +13,71 @@
  *
  * @author allen
  */
-public final class Matrix {
+public final class Matrix implements DefinableAdjacent{
 
-    private final Row firstRow;
-    private final Row secondRow;
-    private final Row thirdRow;
+    private List<Row> rows;
+    private final int NUMBER_OF_ROWS = 3;
     public static List<Vertex> cells;
 
-
     public Matrix() {
         Matrix.cells = new ArrayList<>();
         Cell.content = 1;
-        this.firstRow = new Row();
-        this.secondRow = new Row();
-        this.thirdRow = new Row();
+        this.rows = new ArrayList<>();
+        for(int i =0; i < NUMBER_OF_ROWS; i++){
+            this.rows.add(new Row());
+        }
         this.defineAdjacent();
     }
 
-    private void defineAdjacent() {
-        this.firstRow.initial.creatingVerticalAdjacent(secondRow.initial);
-        this.secondRow.initial.creatingVerticalAdjacent(thirdRow.initial);
-
-        this.firstRow.center.creatingVerticalAdjacent(secondRow.center);
-        this.secondRow.center.creatingVerticalAdjacent(thirdRow.center);
-
-        this.firstRow.last.creatingVerticalAdjacent(secondRow.last);
-        this.secondRow.last.creatingVerticalAdjacent(thirdRow.last);
-        
+    public void defineAdjacent() {
+        for(int i = 0; i < NUMBER_OF_ROWS - 1; i++){
+            Row upRow = rows.get(i);
+            Row downRow = rows.get(i + 1);
+            for(int j = 0; j < upRow.cells.size();j++){
```

Comment:
Boa tarde, 

Nossa ideia é utilizar POO para fugir destes blocos aninhados de for...
o que são i e j ?